src/utils.ts

```diff
@@ -135,6 +137,40 @@ function words<T extends string>(sentence: T): Words<T> {
     .split(/\s+/g) as Words<T>
 }
 
+/**
+ * Truncate a string if it's longer than the given maximum length.
+ * The last characters of the truncated string are replaced with the omission string which defaults to "...".
+ */
+type Truncate<
+  T extends string,
+  Size extends number,
+  Omission extends string = '...',
+> = Math.IsNegative<Size> extends true
+  ? Omission
+  : Math.Subtract<Length<T>, Size> extends 0
+  ? T
+  : Join<[Slice<T, 0, Math.Subtract<Size, Length<Omission>>>, Omission]>
+
+/**
+ * A strongly typed function to truncate a string if it's longer than the given maximum string length.
+ * The last characters of the truncated string are replaced with the omission string which defaults to "...".
+ * @param sentence the sentence to extract the words from.
+ * @param length the maximum length of the string.
+ * @param omission the string to append to the end of the truncated string.
+ * @returns the truncated string
+ * @example truncate('Hello, World', 8) // 'Hello...'
+ */
+function truncate<T extends string, S extends number, P extends string = '...'>(
+  sentence: T,
+  length: S,
+  omission = '...' as P,
+): Truncate<T, S, P> {
+  if (length <= 0) return omission as Truncate<T, S, P>
```

Comment:
Another lil adjustment to have the exact same implementation
```suggestion
  if (length < 0) return omission as Truncate<T, S, P>
```

Reply:
done in https://github.com/gustavoguichard/string-ts/commit/14bebeb5be0723e0e0a8362b459b92087424afa3